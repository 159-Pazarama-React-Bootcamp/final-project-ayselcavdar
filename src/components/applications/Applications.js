import { useState } from 'react';
import styles from './Applications.module.css';
import { Link } from 'react-router-dom';
import Button from '../button/Button';
import ApplicationCard from './ApplicationCard';
import ThanksMessage from './ThanksMessage';

const Applications = () => {
  const [applications] = useState([
    {
      id: 0,
      queryCode: '000000',
      name: 'obi',
      reason: 'sadasdasdasdsd',
      date: '10.11.2021',
    },
    {
      id: 1,
      queryCode: '111111',
      name: 'obi',
      reason: 'sadasdasdasdsd',
      date: '10.11.2021',
    },
    {
      id: 2,
      queryCode: '222222',
      name: 'obi',
      reason: 'sadasdasdasdsd',
      date: '10.11.2021',
    },
    
   
  ]);
  const [userQueryCode, setUserQueryCode] = useState('');
  const handleQueryCodeChange = (e) => setUserQueryCode(e.target.value);

  return (
    <div className={styles['applications-main--container']}>
      <div className={styles['header-container']}>
        <ThanksMessage />
        <div className={styles['subheader-section']}>
          <div>
            <input
              type="text"
              id="queryCode"
              maxLength="10"
              value={userQueryCode}
              onChange={handleQueryCodeChange}
              placeholder="Başvuru kodunuzu giriniz.."
              className={styles['query-input']}
            />
          </div>
          <div className={styles['btn-container']}>
            <Link to={`/basvurular/${userQueryCode}`}>
              <Button type="button" content={'Sorgula'} />
            </Link>
            <div className={styles['new-app-btn']}>
              <Link to={'/'}>
                <Button type="reset" content={'Yeni Basvuru Oluştur'}/>
              </Link>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div className={styles['appList-container']}>
          {applications
            .filter((item) => item.queryCode.includes(userQueryCode))
            .map((app) => (
              <ApplicationCard app={app} key={app.id} title="card title" />
            ))}
        </div>
      </div>
    </div>
  );
};

export default Applications;
