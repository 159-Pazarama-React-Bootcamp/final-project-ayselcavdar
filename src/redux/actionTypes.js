const LOGIN_START = 'LOGIN_START';
const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
const LOGIN_FAIL = 'LOGIN_FAIL';

const LOGOUT_START = 'LOGOUT_START';
const LOGOUT_SUCCESS = 'LOGOUT_SUCCESS';
const LOGOUT_FAIL = 'LOGOUT_FAIL';

const SET_USER = 'SET_USER';

const GET_APPLICATIONS_START = 'GET_APPLICATIONS_START';
const GET_APPLICATIONS_SUCCESS = 'GET_APPLICATIONS_SUCCESS';
const GET_APPLICATIONS_FAIL = 'GET_APPLICATIONS_FAIL';

const DELETE_APPLICATION_START = 'DELETE_APPLICATION_START';
const DELETE_APPLICATION_FAIL = 'DELETE_APPLICATION_FAIL';
const DELETE_APPLICATION_SUCCESS = 'DELETE_APPLICATION_SUCCESS';

const CREATE_APPLICATION_START = 'CREATE_APPLICATION_START';
const CREATE_APPLICATION_SUCCESS = 'CREATE_APPLICATION_SUCCESS';
const CREATE_APPLICATION_FAIL = 'CREATE_APPLICATION_FAIL';

const EDIT_APPLICATION_START = 'EDIT_APPLICATION_START';
const EDIT_APPLICATION_FAIL = 'EDIT_APPLICATION_FAIL';
const EDIT_APPLICATION_SUCCESS = 'EDIT_APPLICATION_SUCCESS';


export {
  LOGIN_START,
  LOGIN_SUCCESS,
  LOGIN_FAIL,
  LOGOUT_START,
  LOGOUT_SUCCESS,
  LOGOUT_FAIL,
  SET_USER,
  CREATE_APPLICATION_FAIL,
  CREATE_APPLICATION_START,
  CREATE_APPLICATION_SUCCESS,
  GET_APPLICATIONS_FAIL,
  GET_APPLICATIONS_START,
  GET_APPLICATIONS_SUCCESS,
  DELETE_APPLICATION_START,
  DELETE_APPLICATION_SUCCESS,
  DELETE_APPLICATION_FAIL,
  EDIT_APPLICATION_FAIL,
  EDIT_APPLICATION_START,
  EDIT_APPLICATION_SUCCESS,
};
