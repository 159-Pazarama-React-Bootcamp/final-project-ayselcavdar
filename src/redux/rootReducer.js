import {combineReducers} from 'redux';
import { authReducer } from './reducers/authReducer';
import applicationReducer from './reducers/applicationReducer';

const reducers = {
  user: authReducer,
  data: applicationReducer,
};

export const rootReducer = combineReducers(reducers);