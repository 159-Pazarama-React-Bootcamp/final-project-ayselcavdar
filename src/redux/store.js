import {createStore, applyMiddleware} from 'redux';
import {rootReducer} from './rootReducer';
import logger from 'redux-logger';
import thunk from 'redux-thunk';

const middleware = [thunk];
// state değişikliklerini takibe yarayan anlık loglar sunan logger modulü,
// yalnızca development ortamı için uygulanmıştır
if(process.env.NODE_ENV === 'development'){
    middleware.push(logger);
}
export const store = createStore(rootReducer, applyMiddleware(...middleware));