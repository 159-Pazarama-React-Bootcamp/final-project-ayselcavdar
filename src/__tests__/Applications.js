/**
 * @jest-environment jsdom
 */
import render from 'react-test-renderer';
import Applications from '../components/applications/Applications';
import { Provider } from 'react-redux';
import { store } from '../redux/store';
import { BrowserRouter } from 'react-router-dom';


describe('<Applications /> rendering', () => {
  it('AppDetails snapshots check', () => {
    const wrapper = render.create(
      <Provider store={store}>
        <BrowserRouter>
          <Applications />
        </BrowserRouter>
      </Provider>
    );
    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
