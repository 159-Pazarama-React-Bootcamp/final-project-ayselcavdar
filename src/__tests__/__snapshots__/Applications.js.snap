// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Applications /> rendering AppDetails render check 1`] = `
<div
  className="applications-main--container"
>
  <div
    className="header-container"
  >
    <div
      className="message-container"
    >
      <p>
        Teşekkürler! Başvurunuz başarıyla alınmıştır.
      </p>
      <p>
        Başvuru durumunuzu detaya tıklayarak öğrenebilirsiniz.
      </p>
    </div>
    <div
      className="subheader-section"
    >
      <div>
        <input
          className="query-input"
          id="queryCode"
          maxLength="10"
          name="queryCode"
          onChange={[Function]}
          placeholder="Başvuru kodunuz ile filtreleyiniz.."
          type="text"
          value=""
        />
        <div
          className="form-group--select "
        >
          <select
            className="form-control"
            name="filterVal"
            onChange={[Function]}
            placeholder=""
            value=""
          >
            <option
              value=""
            >
              Başvuru durumuna göre filtereleyiniz
            </option>
            <option
              value="EVALUATING"
            >
              EVALUATING
            </option>
            <option
              value="ANSWERED"
            >
              ANSWERED
            </option>
          </select>
        </div>
      </div>
      <div
        className="btn-container"
      >
        <div
          className="new-app-btn"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <button
              className="btn"
              style={
                Object {
                  "backgroundColor": "#E6E6E6",
                  "color": "#6B63FF",
                }
              }
              type="reset"
            >
              Yeni Basvuru Oluştur
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="appList-container"
    >
      <div
        className="alert-container alert-info "
      >
        <div
          className="title-container"
        >
          <h3
            className="title-text"
          >
            Bilgilendirme
          </h3>
        </div>
        <div
          className="body-container"
        >
          <p
            className="body-text"
          >
            Herhangi bir başvuru bulunamadı.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Applications /> rendering AppDetails snapshots check 1`] = `
<div
  className="applications-main--container"
>
  <div
    className="header-container"
  >
    <div
      className="message-container"
    >
      <p>
        Teşekkürler! Başvurunuz başarıyla alınmıştır.
      </p>
      <p>
        Başvuru durumunuzu detaya tıklayarak öğrenebilirsiniz.
      </p>
    </div>
    <div
      className="subheader-section"
    >
      <div>
        <input
          className="query-input"
          id="queryCode"
          maxLength="10"
          name="queryCode"
          onChange={[Function]}
          placeholder="Başvuru kodunuz ile filtreleyiniz.."
          type="text"
          value=""
        />
        <div
          className="form-group--select "
        >
          <select
            className="form-control"
            name="filterVal"
            onChange={[Function]}
            placeholder=""
            value=""
          >
            <option
              value=""
            >
              Başvuru durumuna göre filtereleyiniz
            </option>
            <option
              value="EVALUATING"
            >
              EVALUATING
            </option>
            <option
              value="ANSWERED"
            >
              ANSWERED
            </option>
          </select>
        </div>
      </div>
      <div
        className="btn-container"
      >
        <div
          className="new-app-btn"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <button
              className="btn"
              style={
                Object {
                  "backgroundColor": "#E6E6E6",
                  "color": "#6B63FF",
                }
              }
              type="reset"
            >
              Yeni Basvuru Oluştur
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="appList-container"
    >
      <div
        className="alert-container alert-info "
      >
        <div
          className="title-container"
        >
          <h3
            className="title-text"
          >
            Bilgilendirme
          </h3>
        </div>
        <div
          className="body-container"
        >
          <p
            className="body-text"
          >
            Herhangi bir başvuru bulunamadı.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AppDetails render check 1`] = `
<div
  className="applications-main--container"
>
  <div
    className="header-container"
  >
    <div
      className="message-container"
    >
      <p>
        Teşekkürler! Başvurunuz başarıyla alınmıştır.
      </p>
      <p>
        Başvuru durumunuzu detaya tıklayarak öğrenebilirsiniz.
      </p>
    </div>
    <div
      className="subheader-section"
    >
      <div>
        <input
          className="query-input"
          id="queryCode"
          maxLength="10"
          name="queryCode"
          onChange={[Function]}
          placeholder="Başvuru kodunuz ile filtreleyiniz.."
          type="text"
          value=""
        />
        <div
          className="form-group--select "
        >
          <select
            className="form-control"
            name="filterVal"
            onChange={[Function]}
            placeholder=""
            value=""
          >
            <option
              value=""
            >
              Başvuru durumuna göre filtereleyiniz
            </option>
            <option
              value="EVALUATING"
            >
              EVALUATING
            </option>
            <option
              value="ANSWERED"
            >
              ANSWERED
            </option>
          </select>
        </div>
      </div>
      <div
        className="btn-container"
      >
        <div
          className="new-app-btn"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <button
              className="btn"
              style={
                Object {
                  "backgroundColor": "#E6E6E6",
                  "color": "#6B63FF",
                }
              }
              type="reset"
            >
              Yeni Basvuru Oluştur
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="appList-container"
    >
      <div
        className="alert-container alert-info "
      >
        <div
          className="title-container"
        >
          <h3
            className="title-text"
          >
            Bilgilendirme
          </h3>
        </div>
        <div
          className="body-container"
        >
          <p
            className="body-text"
          >
            Herhangi bir başvuru bulunamadı.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
